### ISSUE TITLE
Bad result for tutorial.mnist_blackbox

### ISSUE THREAD

**Issue by leviswind** [tensorflow/cleverhans#183]
I run the tutorial code called mnist_blackbox.py, however the result is not the expected for the test accuracy of oracle on adversarial generated using the substitute is 0.85096. It  seems the adversarial examples have little effect on the test accuracy. Has anyone else meet this problem? By the way, the version of tensorflow I used is 1.2.1. I think it's not the problem of tensorflow.

```
Successfully downloaded train-images-idx3-ubyte.gz 9912422 bytes.
Extracting /tmp/train-images-idx3-ubyte.gz
Successfully downloaded train-labels-idx1-ubyte.gz 28881 bytes.
Extracting /tmp/train-labels-idx1-ubyte.gz
Successfully downloaded t10k-images-idx3-ubyte.gz 1648877 bytes.
Extracting /tmp/t10k-images-idx3-ubyte.gz
Successfully downloaded t10k-labels-idx1-ubyte.gz 4542 bytes.
Extracting /tmp/t10k-labels-idx1-ubyte.gz
X_train shape: (60000, 28, 28, 1)
X_test shape: (10000, 28, 28, 1)
Preparing the black-box model.
Defined TensorFlow model graph.
Completed model training.
Test accuracy of black-box on legitimate test examples: 0.989746192155
Training the substitute model.
Defined TensorFlow model graph for the substitute.
Substitute training epoch #0
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #1
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #2
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #3
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #4
Completed model training.
Augmenting substitute training data.
Labeling substitute training data.
Substitute training epoch #5
Completed model training.
Test accuracy of oracle on adversarial examples generated using the substitute: 0.850964466594
```

**Comment by abhibhav14** [comment]
Yes, there is a related issue raised #176 . Switched to the Adam optimizer has resulted this drop in accuracy. Using Adadelta will get you better results, a test accuracy of 0.497 on the adversarial samples. Something will be done about this soon.

**Comment by catherio** [comment]
@npapernot What's the plan to address this issue? It looks like #176 didn't address this?

**Comment by npapernot** [comment]
I made a note to take another look a few days ago but I did not get a chance to look at it yet.

**Comment by goodfeli** [comment]
@npapernot : was this fixed by #322 ?

**Comment by npapernot** [comment]
Yes, I might rewrite the tutorial later to show how transferability is affected by the different architecture choices (e.g., including convolutions vs. not including any). But I have a note in my todo list to do this, it's not necessary to keep the issue open.
